expression/builtin_time.go

```diff
@@ -665,41 +665,32 @@ func (c *monthNameFunctionClass) getFunction(args []Expression, ctx context.Cont
 	if err := c.verifyArgs(args); err != nil {
 		return nil, errors.Trace(err)
 	}
-	sig := &builtinMonthNameSig{newBaseBuiltinFunc(args, ctx)}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpTime)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf.tp.Flen = 10
+	sig := &builtinMonthNameSig{baseStringBuiltinFunc{bf}}
 	return sig.setSelf(sig), nil
 }
 
 type builtinMonthNameSig struct {
-	baseBuiltinFunc
-}
-
-func (b *builtinMonthNameSig) eval(row []types.Datum) (d types.Datum, err error) {
-	args, err := b.evalArgs(row)
-	if err != nil {
-		return d, errors.Trace(err)
-	}
-	return builtinMonthName(args, b.ctx)
+	baseStringBuiltinFunc
 }
 
-// builtinMonthName ...
-// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_monthname
-func builtinMonthName(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
-	d, err = builtinMonth(args, ctx)
-	if err != nil || d.IsNull() {
-		return d, errors.Trace(err)
+func (b *builtinMonthNameSig) evalString(row []types.Datum) (d string, isNull bool, err error) {
```

Comment:
`(d string, isNull bool, err error)` is ugly, maybe `(string, isNull, error)` is better ? or a more readable variable name ?